### PR TITLE
Making stores async

### PR DESCRIPTION
**Pull request by rmachielse**
Hi @marcoow,

Here is the pull request on #602 we discussed some time ago, to make stores async by default.
This will allow people to build stores that persist on asynchronous storage (like chrome app storage).

You mentioned that you would like to postpone this to release 1.1, but this change might be major, since everyone with a custom stores needs to adapt to the asynchronous workflow.

### CODE REVIEW

**Comment by marcoow** [comment]
This is great, thanks!

I still think this can be merged without requiring a new major rebase though. Basically all that's needed is a method in the internal session that wraps all store access into a Promise if the store doesn't return a promise anyway. 1.0 has been delayed so much already it really needs to get released now. Of course making this PR work with non-async stores will require some work but I think it's worth it to not delay 1.0 any longer.

Definitely going to merge this post 1.0!

**Comment by rmachielse** [comment]
Alright, I'll add backward compatibility soon. Thanks for the help!

**Comment by marcoow** [comment]
thanks for the effort!

File: addon/session-stores/cookie.js [hunk 1]
```diff
@@ -159,33 +172,43 @@ export default BaseStore.extend({
...
+          this._lastData = data;
+          this.trigger('sessionDataUpdated', data);
+        }
+        if (!Ember.testing) {
+          Ember.run.cancel(this._syncDataTimeout);
+          this._syncDataTimeout = Ember.run.later(this, this._syncData, 500);
+        }
+        resolve();
+      });
+    });
```
Comment (marcoow) [inline addon/session-stores/cookie.js]: Does this have to return a promise as well? It's actually synchronous and doesn't belong to the public API (which is only `persist`, `restore`, `clear`).

Comment (marcoow) [inline addon/session-stores/cookie.js]: forget that - I see how it all makes sense now ;)

**Comment by rmachielse** [comment]
Should I open a new pr on master?

**Comment by marcoow** [comment]
yes please - thanks!

**Pull request closed without merging**